### PR TITLE
Adding Module32First, Module32Next and Module32Enumerate

### PR DESCRIPTION
**Pull request by hmemcpy**
And corresponding structs. This complements exising Process\* methods.

### CODE REVIEW

File: src/Kernel32.Desktop/Kernel32+MODULEENTRY32.cs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                        return new string(exePath);
+                    }
+                }
+            }
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="MODULEENTRY32" /> struct
+            /// with <see cref="dwSize" /> set to the correct value.
+            /// </summary>
+            /// <returns>An instance of <see cref="PROCESSENTRY32"/>.</returns>
```
Comment (vbfox) [inline src/Kernel32.Desktop/Kernel32+MODULEENTRY32.cs]: `PROCESSENTRY32` -> `MODULEENTRY32`

Comment (hmemcpy) [inline src/Kernel32.Desktop/Kernel32+MODULEENTRY32.cs]: FOILED BY COPYPASTE!!1 :dancer: 
Will fix, thanks @vbfox

**Comment by vbfox** [comment]
:+1: LGTM

File: src/Kernel32.Desktop/Kernel32+MODULEENTRY32.cs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]
+        [SuppressMessage(
+            "StyleCop.CSharp.MaintainabilityRules",
+            "SA1401:Fields must be private",
+            Justification = "Used in DllImport Marshaling.")]
+        [OfferIntPtrPropertyAccessors]
+        public unsafe partial struct MODULEENTRY32
+        {
+            /// <summary>
+            /// The size of the structure, in bytes. Set automatically by the constructor.
```
Comment (AArnott) [inline src/Kernel32.Desktop/Kernel32+MODULEENTRY32.cs]: Rather than "by the constructor" perhaps you should say: `by the <see cref="Create"/> method.`

**Comment by AArnott** [comment]
Please review your build results either locally or on the appveyor validation and clean up all build warnings.

**Comment by AArnott** [comment]
Thanks, @hmemcpy! You've obviously taken care to follow our patterns closely. :)

We'll be happy to merge after the couple issues I raised are corrected.

**Comment by hmemcpy** [comment]
Done! I was left with 2 xmldoc warnings that the params weren't documented, but they are!
So... no idea :)

**Comment by AArnott** [comment]
> I was left with 2 xmldoc warnings that the params weren't documented, but they are!
> So... no idea :)

Your `</summary>` is in the wrong place for the method generating the warnings.

**Pull request merged**